docs/src/user-guides/shamir-secret-sharing.md

```diff
@@ -0,0 +1,155 @@
+# Shamir Secret Sharing
+
+Neptune Core supports Shamir secret sharing to distribute shares in the wallet secret.
+
+## How It Works
+
+A \\(t\\)-out-of-\\(n\\) Shamir secret sharing scheme works as follows. Let \\(S \in \mathbb{F}\\) be the original secret. In the source code, we use `XFieldElement` as the field \\(\mathbb{F}\\) and `SecretKeyMaterial` as a wrapper around `XFieldElement`s when they are used for this purpose.
+
+Sample a univariate polynomial \\(f(X)\\) of degree at most \\(t-1\\) uniformly at random except for the constant coefficient. Choose \\(S\\) for the constant coefficient, so that \\(f(0) = S\\).
+
+With an implicit embedding \\(\mathbb{N} \rightarrow \mathbb{F}\\) we can associate the \\(i\\)th share with the point \\((i, f(i))\\). Note that \\(i=0\\) is disallowed since \\((0, f(0))\\) corresponds to the secret. To generate \\(n\\) shares we let \\(i\\) range from \\(1\\) to \\(n\\) (including the upper bound).
+
+To reconstruct the original secret it suffices to have *any* \\(t\\) secret shares. Just reconstruct the polynomial and evaluate it at \\(0\\).
+
+However, any selection of *fewer than \\(t\\)* secret shares contains *no information* about the original secret.
+
+## How to Use It
+
+First, make sure you have a wallet installed.
+
+ - Whenever you run `neptune-core`, it will read the wallet file or create one if none is found. Unless you moved or removed this file, it is still there.
+ - To test if the wallet file is present, run `neptune-cli which-wallet`.
+ - To generate a wallet file without running `neptune-core`, try `neptune-cli generate-wallet`.
+ - To import a wallet from a seed phrase, first make sure there is no wallet file, and then run `neptune-cli import-seed-phrase`.
+
+To generate \\(n\\) shares in a \\(t\\)-out-of-\\(n\\) scheme, run `neptune-cli shamir share t n` and replace `t` and `n` with the values you want. This command generates \\(n\\) seed phrases. **Note:** be sure to record the share index ("`i/n`") along with each share, as you will need this information to reconstruct the original secret.
```

Comment:
`neptune-cli shamir share t n` should be `neptune-cli shamir-share t n`  (there's a dash `-` missing).

Reply:
:+1: